contracts/token/ERC20/extensions/ERC20Permit.sol

```diff
@@ -24,6 +24,16 @@ abstract contract ERC20Permit is ERC20, IERC20Permit, EIP712, Nonces {
     bytes32 private constant _PERMIT_TYPEHASH =
         keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
 
+    /**
+     * @dev Permit deadline has expired.
+     */
+    error ERC2612ExpiredDeadline(uint256 deadline);
+
+    /**
+     * @dev Mismatched signature.
+     */
+    error ERC2612InvalidSignature(address signer, address owner);
```

Comment:
I would rename this to `ERC2612InvalidSigner(address signer, address owner)` for consistency with `MinimalForwarder` and future signature validation errors that expect the signature to come from a specific signer.

Note that `ECDSAInvalidSignature()` serves a different purpose. The signature should never recover to `address(0)`, so it is invalid.